bioscrape/inference_setup.py

```diff
@@ -168,35 +134,37 @@ def set_N_simulations(self, N_simulations: int):
 
     def set_initial_conditions(self, initial_conditions):
         '''
-        Set the initial_conditions for parameter inference. 
-        Must be a dictionary object with keys pointing to the species measurements (in order of self.measurements).
-        Values of dictionary may be list of different initial conditions for which the data was collected. 
+        Set the initial conditions for parameter inference. 
+        Must be a dictionary object with keys pointing 
+        to the species measurements (in order of self.measurements).
+        The dictionary value is the initial condition. 
+        A list of such dictionary may be passed
+        in corresponding to each measurement. 
         '''
-        if type(initial_conditions) is dict:
-            if len(list(initial_conditions.keys())) != len(self.M.get_species_array()):
-                new_ic = dict(self.M.get_species_dictionary())
-                for key, value in initial_conditions.items():
-                    new_ic[key] = value
-                self.initial_conditions = new_ic
-            else:
-                self.initial_conditions = initial_conditions
-        elif type(initial_conditions) is list and len(initial_conditions):
+        # Get initial_conditions from the model if not given explicitly
+        if initial_conditions is None: 
+            initial_conditions = self.M.get_species_dictionary()
+        if type(initial_conditions) is list and len(initial_conditions):
             for curr_ic_index in range(len(initial_conditions)):
                 ic = initial_conditions[curr_ic_index]
                 if type(ic) is not dict:
                     raise ValueError('All entries in the initial condition list must be dictionaries.')
-                elif len(list(ic.keys())) != len(self.M.get_species_array()):
-                    new_ic = dict(self.M.get_species_dictionary())
-                    for key, value in ic.items():
-                        new_ic[key] = value
-                    initial_conditions[curr_ic_index] = new_ic
-            self.initial_conditions = initial_conditions
-        # Get initial_conditions from the model if not given explicitly
-        initial_conditions = self.initial_conditions
-        if initial_conditions is None: 
-            self.initial_conditions = self.M.get_species_dictionary()
+        self.initial_conditions = initial_conditions
+        return True 
+
+    def set_parameter_conditions(self, parameter_conditions):
+        '''
+        Set the parameter conditions for parameter inference. 
+        Must be a dictionary object with keys pointing to the 
+        parameters that change with each measurement
+        Value of dictionary is the parameter value. 
+        A list of such dictionaries may be passed in 
+        corresponding to each measurement.
+        '''
+        self.parameter_conditions = parameter_conditions
```

Comment:
Do we need any checks here to ensure that the parameter conditions are the correct size/shape? What about if the parameters being set are also parameters being inferred?